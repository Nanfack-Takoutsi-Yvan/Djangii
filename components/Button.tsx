/* eslint-disable react/require-default-props */
import {
  TouchableOpacity,
  Text,
  StyleSheet,
  StyleProp,
  ViewStyle,
  Dimensions
} from "react-native"

type ButtonProps = {
  text: string
  color?: string
  style?: StyleProp<ViewStyle>
}

export default function Button({ text, style, color }: ButtonProps) {
  return (
    <TouchableOpacity
      activeOpacity={0.8}
      // eslint-disable-next-line no-console
      onPress={() => console.log("pressed")}
      style={[styles.btn, style]}
    >
      <Text
        style={{
          ...styles.buttonText,
          color
        }}
      >
        {text}
      </Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  btn: {
    height: 50,
    borderRadius: 20,
    justifyContent: "center",
    alignItems: "center",
    paddingHorizontal: 56,
    marginVertical: 10
  },
  buttonText: {
    fontSize: 0.035 * Dimensions.get("window").width,
    fontFamily: "SoraSemibold"
  }
})
