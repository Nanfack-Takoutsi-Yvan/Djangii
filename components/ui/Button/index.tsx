/* eslint-disable react/require-default-props */
import {
  TouchableOpacity,
  Text,
  StyleSheet,
  StyleProp,
  ViewStyle,
  Dimensions
} from "react-native"
import { Icon, IconComponentProvider } from "@react-native-material/core"
import MaterialCommunityIcons from "@expo/vector-icons/MaterialCommunityIcons"

type ButtonProps = {
  text: string
  color?: string
  iconSize?: number
  iconRight?: string
  iconColor?: string
  style?: StyleProp<ViewStyle>
  OnPress?: () => void
}

export default function Button({
  text,
  style,
  color,
  iconRight,
  iconSize,
  iconColor,
  OnPress
}: ButtonProps) {
  return (
    <IconComponentProvider IconComponent={MaterialCommunityIcons}>
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={OnPress}
        style={[styles.btn, style]}
      >
        <Text
          style={{
            ...styles.buttonText,
            color
          }}
        >
          {text}
        </Text>
        {iconRight && (
          <Icon name={iconRight} size={iconSize || 24} color={iconColor} />
        )}
      </TouchableOpacity>
    </IconComponentProvider>
  )
}

const styles = StyleSheet.create({
  btn: {
    height: 50,
    borderRadius: 20,
    justifyContent: "center",
    alignItems: "center",
    paddingHorizontal: 56,
    marginVertical: 10,
    flexDirection: "row",
    columnGap: 12
  },
  buttonText: {
    fontSize: 0.035 * Dimensions.get("window").width,
    fontFamily: "SoraSemibold"
  }
})
