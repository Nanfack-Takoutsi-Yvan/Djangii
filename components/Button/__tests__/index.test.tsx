import * as React from "react"
import renderer from "react-test-renderer"

import Button from "../index"

it(`renders correctly`, () => {
  const tree = renderer.create(<Button text="Hello world" />).toJSON()

  expect(tree).toMatchSnapshot()
})
