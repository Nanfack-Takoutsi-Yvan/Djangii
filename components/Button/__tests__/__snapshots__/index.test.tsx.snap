// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderRadius": 20,
      "columnGap": 12,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "marginVertical": 10,
      "opacity": 1,
      "paddingHorizontal": 56,
    }
  }
>
  <Text
    style={
      {
        "color": undefined,
        "fontFamily": "SoraSemibold",
        "fontSize": 26.250000000000004,
      }
    }
  >
    Hello world
  </Text>
</View>
`;
